packages/editor/src/store/reducer.js

```diff
@@ -279,259 +267,270 @@ export const editor = flow( [
 		return state;
 	},
 
-	blocksByClientId( state = {}, action ) {
-		switch ( action.type ) {
-			case 'RESET_BLOCKS':
-			case 'SETUP_EDITOR_STATE':
-				return getFlattenedBlocks( action.blocks );
+	blocks: flow( [
+		combineReducers,
+
+		// Track whether changes exist, resetting at each post save. Relies on
+		// editor initialization firing post reset as an effect.
+		withChangeDetection( {
+			resetTypes: [ 'SETUP_EDITOR_STATE', 'REQUEST_POST_UPDATE_START' ],
+			ignoreTypes: [ 'RECEIVE_BLOCKS', 'RESET_POST', 'UPDATE_POST' ],
+		} ),
+	] )( {
```

Comment:
Is everything below here just indented? It's a bit hard to see the code changes, if there are any.

Reply:
>Is everything below here just indented? It's a bit hard to see the code changes, if there are any.

Yes, no changes aside from indentation. 

GitHub has an option to hide whitespace-only changes, which should help make things a bit clearer.

![image](https://user-images.githubusercontent.com/1779930/47292885-650f0180-d5d6-11e8-8100-be05f45b4987.png)

Reply:
Funny that this Higher order reducer moved from "edits" to just "blocks". It almost feels like we need an `editBlocks` :)